File: src/patterns/structural/adapter/classadapter.nim
# std
import
    std/[
        json
    ]

# internal
import
    patterns/structural/adapter/[
        government
    ]


type
    ClassAdapter* = ref object of Government


proc newClassAdapter*(): ClassAdapter =
    return ClassAdapter()

proc sendGainsInfo*(self: ClassAdapter, data: JsonNode): bool =
    return self.sendGainsInformation((data["taxPayerId"].getInt().uint(), data["taxPayerGains"].getFloat()))

File: src/patterns/structural/adapter/classadapter.go
package adapter

import (
	"encoding/json"
)


type ClassAdapter struct {
	*Government
}


func NewClassAdapter() *ClassAdapter {
	return &ClassAdapter{
		&Government{},
	}
}

func (self *ClassAdapter) SendGainsInfo(data []byte) bool {
	var jsonValues map[string]any
	json.Unmarshal(data, &jsonValues)
	return self.SendGainsInformation(Tuple{TaxPayerId: jsonValues["tax_payer_id"], TaxPayerGains: jsonValues["tax_payer_gains"]})
}